machine/v1beta1/types_awsprovider.go

```diff
@@ -160,6 +165,27 @@ type SpotMarketOptions struct {
 	MaxPrice *string `json:"maxPrice,omitempty"`
 }
 
+type InstanceMetadataHTTPTokens string
+
+const (
+	// InstanceMetadataHTTPTokensRequired enforces sending of a signed token header with any instance metadata retrieval (GET) requests.
+	// Enforces IMDSv2 usage.
+	InstanceMetadataHTTPTokensRequired = "required"
+	// InstanceMetadataHTTPTokensOptional allows IMDSv1 usage along with IMDSv2
+	InstanceMetadataHTTPTokensOptional = "optional"
+)
+
+// MetadataServiceOptions defines the options available to a user when configuring
+// Instance Metadata Service (IMDS) Options.
+type MetadataServiceOptions struct {
+	// HTTPTokens determines whether or not the host requires the use of authentication when interacting with the metadata service.
+	// When using authentication, this enforces v2 interaction method (IMDSv2) with the metadata service.
+	// When omitted, this means the user has no opinion and the value is left to the platform to choose a good
```

Comment:
Some of the message got lost here
```suggestion
	// When omitted, this means the user has no opinion and the value is left to the platform to choose a good
	// default, which is subject to change over time. The current default is optional.
```